src/v/cluster/data_policy_manager.cc

```diff
@@ -24,16 +24,34 @@ namespace cluster {
 
 namespace {
 
+bool topic_exists(
+  const ss::sharded<topic_table>& tt, const model::topic_namespace& tn) {
+    auto& topics = tt.local().topics_map();
+    auto tp = topics.find(tn);
+    return tp != topics.end();
+}
+
 template<typename Cmd>
-std::error_code do_apply(Cmd cmd, data_policy_manager::container_type& db) {
+std::error_code do_apply(
+  Cmd cmd,
+  data_policy_manager::container_type& db,
+  ss::sharded<topic_table>& topic_table) {
     return ss::visit(
       std::move(cmd),
-      [&db](create_data_policy_cmd cmd) {
+      [&db, &topic_table](create_data_policy_cmd cmd) {
+          if (!topic_exists(topic_table, cmd.key)) {
```

Comment:
I think that this behavior should be moved to a separate PR that we add later and discuss. It isn't required for initial scope of work, and I think the policy itself is a little restricting because it creates a tight coupling between topics and data policies, especially on the order in which operations are run against the cluster (ie topic first, then policy).

Definitely don't throw away these patches and keep them in a separate branch so we can easily grab them back later